torchrl/objectives/ppo.py

```diff
@@ -466,12 +471,25 @@ def _log_weight(
             self.actor_network
         ) if self.functional else contextlib.nullcontext():
             dist = self.actor_network.get_dist(tensordict)
-        log_prob = dist.log_prob(action)
 
         prev_log_prob = tensordict.get(self.tensor_keys.sample_log_prob)
         if prev_log_prob.requires_grad:
             raise RuntimeError("tensordict prev_log_prob requires grad.")
 
+        if isinstance(dist, CompositeDistribution):
+            if (
+                tensordict.get(self.tensor_keys.action).batch_size
+                != tensordict.batch_size
+            ):
+                # This condition can be True in notensordict usage
+                tensordict.get(
+                    self.tensor_keys.action
+                ).batch_size = tensordict.batch_size
```

Comment:
get operations do not come for free, if you're accessing the same entity twice (eg action) I would do it just once before.